lms/lmsweb/views.py

```diff
@@ -396,22 +412,26 @@ def user(user_id):
 
     return render_template(
         'user.html',
-        solutions=Solution.of_user(target_user.id, with_archived=True),
+        solutions=Solution.of_user(
+            target_user.id, with_archived=True, select_all=True,
+        ),
         user=target_user,
         is_manager=is_manager,
         notes_options=Note.get_note_options(),
     )
 
 
-@webapp.route('/send', methods=['GET'])
+@webapp.route('/send/<int:course_id>', methods=['GET'])
 @login_required
-def send_():
-    return render_template('upload.html')
+def send_(course_id: int):
+    if not UserCourse.is_user_registered(current_user.id, course_id):
+        return fail(403, "You aren't allowed to watch this page.")
+    return render_template('upload.html', course_id=course_id)
 
 
-@webapp.route('/upload', methods=['POST'])
+@webapp.route('/upload/<int:course_id>', methods=['POST'])
 @login_required
-def upload_page():
+def upload_page(course_id: int):
```

Comment:
We shall check if the user is registered to this course also here, IMHO

Reply:
We check it in the `models/upload.py` file:
https://github.com/PythonFreeCourse/lms/pull/307/files#diff-3531202a8627fffa785e40202356f4c0f0228778b692178ed976f05d60ca138bR35